### PR TITLE
LPS-164601 | Add Autom. test FDSTable#HeadersCanBeLocalized

### PR DESCRIPTION
**Pull request by deborasoliveira**
**Test Plan**

- Given: localized FDS sample portlet
- And Given: site URL localized to Spanish //localhost:8080/es/*
- When: on classic tab view
- Then: There is no hyphen in any header text
- And Then: First letter is capitalized for all headers
- And Then: localized Spanish headers are present //all headers have been translated to Spanish

**JIRA Ticket:**
https://issues.liferay.com/browse/LPS-164601

### CODE REVIEW

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

**Comment by deborasoliveira** [comment]
The error 'java.lang.Exception: Element is not present at "//body"' is showing again. That's why the test didn't pass.

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

**Comment by deborasoliveira** [comment]
ci:test:sf

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase [hunk 1]
```diff
@@ -49,9 +49,43 @@ definition {
...
 	@priority = "4"
 	test HeadersCanBeLocalized {
+		task ("And Given: site URL localized to Spanish //localhost:8080/es/*") {
+			var portalURL = PropsUtil.get("portal.url");
 
-		// To-do HeadersCanBeLocalized Autom. test
+			Navigator.openSpecificURL(url = "${portalURL}/es/home");
 
+			Navigator.gotoPage(pageName = "Frontend Data Set Test Page");
+		}
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase:58]: Hey Debs could you add `Navigator.gotoPage(pageName = "Frontend Data Set Test Page"); `here? When it changes the language of the page is being directed to home page and this is breaking test. I think could work

Comment (deborasoliveira) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase:58]: before which command?

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by john-co** [comment]
Looks like you already found a workaround, but I think opening the page directly will be more robust:
https://github.com/john-co/liferay-portal/pull/487#issuecomment-1294273504

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

**Comment by deborasoliveira** [comment]
ci:test:sf

**Comment by deborasoliveira** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
Just started reviewing :)

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase [hunk 2]
```diff
@@ -60,9 +60,13 @@ definition {
...
-			AssertTextNotPresent(
-				locator1 = "FrontendDataSet#HEADER_ROW",
-				value1 = "-");
+			var header = selenium.getText("FrontendDataSet#HEADER_ROW");
+
+			var containsHyphen = StringUtil.contains("${header}", "-");
+
+			TestUtils.assertEquals(
+				actual = "${containsHyphen}",
+				expected = "false");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase:69]: This is very interesting. Does this work better to capture all the text in the row than the function AssertTextNotPresent?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase [hunk 3]
```diff
@@ -49,9 +47,44 @@ definition {
...
-		// To-do HeadersCanBeLocalized Autom. test
+			var containsHyphen = StringUtil.contains("${header}", "-");
 
+			TestUtils.assertEquals(
+				actual = "${containsHyphen}",
+				expected = "false");
+		}
+
+		task ("And Then: First letter is capitalized for all headers and localized Spanish headers are present") {
+			AssertTextEquals.assertTextCaseSensitive(
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase:73]: nice, you found and used the case sensitive variation.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path [hunk 4]
```diff
@@ -231,6 +231,11 @@
 	<td>//*[@class="dropdown custom-views-actions"]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>HEADER_ROW</td>
+	<td>xpath=(//div[contains(@class, "dnd-tr")])[1]</td>
+	<td></td>
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:237]: For naming paths, generally I like to relate the name of the path with some attribute within the xpath. In this case, if there's some kind of header attribute or element that I can target that would be better in my opinion. But I understand what this means (first row of a table, which would be the header row).

**Review by john-co — APPROVED** [review]
Looks good to me! :sunglasses:

**Comment by john-co** [comment]
ci:forward

**Comment by john-co** [comment]
ci:forward

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase (hunk 2 above)
Comment (deborasoliveira) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSTable.testcase:69]: Yes, its a workaround for the error related to the assertion of hyphen.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path (hunk 4 above)
Comment (deborasoliveira) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:237]: I'll keep that in mind for next time. I really didn't know how to name this path a clearer name.

Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/frontenddataset/FrontendDataSet.path:237]: I'd prob suggest something like:
```
	<td>TABLE_ROW_1</td>
	<td>xpath=(//div[contains(@class, "dnd-tr")])[1]</td>
```

Other alternatives:
- if we think there'll be usecases to target other rows
```
	<td>TABLE_ROW_INDEX</td>
	<td>xpath=(//div[contains(@class, "dnd-tr")])[${index}]</td>
```
- or refactor this path, we try to have general paths in paths/pathlib that are shared between multiple components:
https://github.com/liferay/liferay-portal/commit/d4e8177b1c83d8e209ea1f51ba39dbd89cea7d4d

**Pull request closed without merging**